src/rust/en.freewebnovel/src/parser.rs

```diff
@@ -0,0 +1,283 @@
+					match index {
+						0 => continue,
+						_ => tag = String::from(tag_list[index]),
+					}
+				} else if filter.name.as_str() == "Sort" {
+					let index = filter.value.as_int().unwrap_or(-1);
+					sort = match index {
+						0 => String::new(),
+						1 => String::from("chinese-novel"),
+						2 => String::from("korean-novel"),
+						3 => String::from("japanese-novel"),
+						4 => String::from("english-novel"),
+						_ => continue,
+					};
+				}
+			}
+			_ => continue,
+		}
+	}
+
+	if !title.is_empty() {
+		let mut mangas: Vec<Manga> = Vec::new();
+		let url = format!("{}/search/", base_url);
+		let request = Request::new(url.as_str(), HttpMethod::Post);
+		let body_data = format!("searchkey={}", title);
+		let html = request
+			.header("X-Requested-With", "XMLHttpRequest")
+			.header("Content-Type", "application/x-www-form-urlencoded")
+			.body(body_data.as_bytes())
+			.html()?;
+
+		for manga in html.select(".ul-list1 .con").array() {
+			let manga_node = manga.as_node()?;
+			let title = manga_node.select("h3").text().read();
+			let id = base_url.clone() + &manga_node.select("a").attr("href").read();
+			let cover = manga_node.select("img").attr("src").read();
+			mangas.push(Manga {
+				id,
+				cover,
+				title,
+				author: String::new(),
+				artist: String::new(),
+				description: String::new(),
+				url: String::new(),
+				categories: Vec::new(),
+				status: MangaStatus::Unknown,
+				nsfw: MangaContentRating::Safe,
+				viewer: MangaViewer::Rtl,
+			});
+		}
+		let last_page = html.select(".pages li").text().read();
+		let has_more = last_page.contains(">>");
+		Ok(MangaPageResult {
+			manga: mangas,
+			has_more,
+		})
+	} else if title.is_empty() && tag.is_empty() && sort.is_empty() {
+		parse_manga_listing(
+			base_url.clone(),
+			format!("{}/latest-release-novel/{}/", base_url, page),
+			String::from("Latest"),
+		)
+	} else {
+		let url = get_filter_url(base_url.clone(), tag, sort, page);
+		parse_manga_listing(base_url, url, String::new())
+	}
+}
+
+pub fn parse_manga_listing(
+	base_url: String,
+	url: String,
+	list_name: String,
+) -> Result<MangaPageResult> {
+	let mut mangas: Vec<Manga> = Vec::new();
+	let html = Request::new(&url, HttpMethod::Get).html()?;
+	for manga in html.select(".ul-list1 .con").array() {
+		let manga_node = manga.as_node()?;
+		let title = manga_node.select("h3").text().read();
+		let id = base_url.clone() + &manga_node.select("a").attr("href").read();
+		let cover = manga_node.select("img").attr("src").read();
+		mangas.push(Manga {
+			id,
+			cover,
+			title,
+			author: String::new(),
+			artist: String::new(),
+			description: String::new(),
+			url: String::new(),
+			categories: Vec::new(),
+			status: MangaStatus::Unknown,
+			nsfw: MangaContentRating::Safe,
+			viewer: MangaViewer::Rtl,
+		});
+	}
+	let last_page = html.select(".pages li").text().read();
+	let has_more = last_page.contains(">>") && list_name != "Popular";
```

Comment:
The last page still has a `>>` in the pagination bar: https://freewebnovel.com/latest-novel/105/

![image](https://user-images.githubusercontent.com/92439990/182048867-4f38193d-df66-4853-89be-cebc571b64b7.png)